src/fec_base.h

```diff
@@ -1007,6 +1022,262 @@ bool FecCode<T>::decode_packet(
+    assert(parities_bufs.size() == n_outputs);
+    assert(parities_props.size() == n_outputs);
+
+    // clear property vectors
+    for (auto& props : parities_props) {
+        props.clear();
+    }
+
+    off_t offset = 0;
+    int block_size = block_size_bytes / word_size;
+
+    // vector of buffers storing data read from chunk
+    vec::Buffers<uint8_t> words_char(n_data, buf_size);
+    const std::vector<uint8_t*> words_mem_char = words_char.get_mem();
+    // vector of buffers storing data that are performed in encoding, i.e. FFT
+    vec::Buffers<T> words(n_data, pkt_size);
+    const std::vector<T*> words_mem_T = words.get_mem();
+
+    int output_len = get_n_outputs();
+
+    // vector of buffers storing data that are performed in encoding, i.e. FFT
+    vec::Buffers<T> output(output_len, pkt_size);
+    const std::vector<T*> output_mem_T = output.get_mem();
+    // vector of buffers storing data in output chunk
+    vec::Buffers<uint8_t> output_char(output_len, buf_size);
+    const std::vector<uint8_t*> output_mem_char = output_char.get_mem();
+
+    reset_stats_enc();
+
+    while (offset < block_size) {
+        size_t remain_size = block_size - offset;
+        size_t copy_size = std::min(pkt_size, remain_size);
+        // TODO: get number of read bytes -> true buf size
+        for (unsigned i = 0; i < n_data; i++) {
+            memcpy(
+                reinterpret_cast<char*>(words_mem_char.at(i)),
+                data_bufs[i] + offset * word_size,
+                copy_size * word_size);
+        }
+
+        vec::pack<uint8_t, T>(
+            words_mem_char, words_mem_T, n_data, copy_size, word_size);
+
+        timeval t1 = tick();
+        uint64_t start = hw_timer();
+        encode(output, parities_props, offset, words);
+        uint64_t end = hw_timer();
+        uint64_t t2 = hrtime_usec(t1);
+
+        total_enc_usec += t2;
+        total_encode_cycles += (end - start) / (copy_size * word_size);
+        n_encode_ops++;
+
+        vec::unpack<T, uint8_t>(
+            output_mem_T, output_mem_char, output_len, pkt_size, word_size);
+
+        for (unsigned i = 0; i < n_outputs; i++) {
+            if (wanted_idxs[i]) {
+                memcpy(
+                    parities_bufs[i] + offset * word_size,
+                    reinterpret_cast<char*>(output_mem_char.at(i)),
+                    copy_size * word_size);
+            }
+        }
+        offset += pkt_size;
+    }
+}
+
+/**
+ * Decode blocks
+ *
+ * @param data_bufs if SYSTEMATIC must be exactly n_data otherwise it is
+ * unused (use nullptr when missing)
+ * @param parities_bufs if SYSTEMATIC must be exactly n_parities otherwise
+ * get_n_outputs() (use nullptr when missing)
+ * @param parities_props if SYSTEMATIC must be exactly n_parities
+ * otherwise get_n_outputs() caller is supposed to provide specific information
+ * bound to parities
+ * @param missing_idxs array of missing_idxs of len code_len indicating presence
```

Comment:
Missing is when the block is physically missing. You may want to reconstruct only one missing fragment (the code below may do less expensive math operations). And also missing applies to both data and parities, while wanted in this case only apply for data.

Reply:
Ok, then we should probably update the docstring, because stating that  `missing_idxs array of missing_idxs ` and `wanted_idxs array of missing_idxs` doens't really help to understand the difference and the meaning of those parameters.